src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+      case 'removeProperties':
+        if (Array.isArray(action.keysToRemove)) {
```

Comment:
The check `Array.isArray(action.keysToRemove)` is good for ensuring that `keysToRemove` is an array before attempting to iterate over it. However, consider moving this check outside of the `removeProperties` case, as it's a form of validation that should happen before any action is applied. If the structure of `action` is not valid, it should be handled before entering the switch statement.